docs/proposals/parameter-distribution.md

```diff
@@ -0,0 +1,85 @@
+# Proposal for Supporting various parameter distributions in Katib
+
+## Summary
+The goal of this project is to enhance the existing Katib Experiment APIs to support various parameter distributions such as uniform, log-uniform, and qlog-uniform. Then extend the suggestion services to be able to configure distributions for search space using libraries provided in each framework.
+
+## Motivation
+Currently, Katib is limited to supporting only uniform distribution for integer, float, and categorical hyperparameters. By introducing additional distributions, Katib will become more flexible and powerful in conducting hyperparameter optimization tasks.
+
+## Proposal
+
+### Maintaining two versions of CRD APIs
+- Introduce `v1beta2` API version
+- The (conversion webhook)[https://book.kubebuilder.io/multiversion-tutorial/conversion.html] will serve as a critical component in managing transitions between different API versions (`v1beta1` to `v1beta2`).
+- TODO 
+> I will create a correspondence table between v1beta1 and v1beta2. Maybe, we only need to create a table for the `ParameterType` and the `FeasibleSpace`.
+Look into this.
+- A specific field will be added to the katib-config `katib/pkg/util/v1beta1/katibconfig/config.go` to handle different gRPC client versions required by the suggestion controller. (One approach could be this for maintaining two versions of gRPC APIs).
+- TODO Based on this comment here:
+https://github.com/kubeflow/katib/pull/2059#discussion_r1049329229
+Look how to maintain two versions of gRPC APIs
+
+### Experiment API changes
+Scope: `pkg/apis/controller/experiments/v1beta1/experiment_types.go`
+- Modifying the `ParameterSpec` struct to include a new field called Distribution.
+
+```go
+type ParameterSpec struct {
+    Name          string       `json:"name,omitempty"`
+    Distribution  Distribution `json:"distribution,omitempty"`
+    FeasibleSpace FeasibleSpace `json:"feasibleSpace,omitempty"`
+}
+```
+
+- Renaming Parameters
+
+```go
+type Distribution string
+
+const (
+ 	CategoricalDistribution     Distribution = "categorical"
+ 	IntUniformDistribution      Distribution = "intUniform"
+ 	IntLogUniformDistribution   Distribution = "intLogUniform"
+ 	FloatUniformDistribution    Distribution = "floatUniform"
+)
```

Comment:
I think, we discussed here that we might don't need to mention the parameter kind (e.g. int and float): https://github.com/kubeflow/katib/pull/2059#discussion_r1046376926